src/CLIFramework/Extension/ExtensionBase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Hook\Hookable;
+use CLIFramework\Hook\HookHolder;
+use CLIFramework\Exception\ExtensionException;
+
+abstract class ExtensionBase
+    implements Extension, Hookable
+{
+    private $hooks;
+
+    private function getHooks()
```

Comment:
Do you mean the `$container` is an instance of `ServiceContainer`? If not so, I've already added an event system to `CommandBase` and `ExtensionBase`. Is the name of `Hook` bad? Is it renamed to `Event`?